owasp-top10-2016-mobile/m5/panda_zap/server/auth/jwt.go

```diff
@@ -0,0 +1,53 @@
+package auth
+
+import (
+	"github.com/dgrijalva/jwt-go"
+	"github.com/labstack/echo"
+	"github.com/spf13/viper"
+	"go.uber.org/zap"
+)
+
+// JWTInterface implements the Auth interface.
+type JWTInterface struct {
+	Logger      *zap.SugaredLogger
+	Settings    *viper.Viper
+	TokenConfig *jwt.Token
+}
+
+// NewJWTSession returns a new JWT session.
+func NewJWTSession(logger *zap.SugaredLogger, settings *viper.Viper) (*JWTInterface, error) {
+	logger.Info("New authentication session created successfully")
+
+	tokenConfig := jwt.New(jwt.SigningMethodHS256)
+
+	authSession := &JWTInterface{
+		Logger:      logger,
+		Settings:    settings,
+		TokenConfig: tokenConfig,
+	}
+
+	return authSession, nil
+}
+
+// NewToken returns a new JWT token
+func (j *JWTInterface) NewToken(id string, username string) (string, error) {
+	claims := j.TokenConfig.Claims.(jwt.MapClaims)
+	claims["id"] = id
+	claims["username"] = username
+
+	token, err := j.TokenConfig.SignedString([]byte(j.Settings.GetString("jwt_secret")))
+	if err != nil {
+		return "", err
+	}
+
+	return token, nil
+}
+
+// GetUser returns the user from the JWT token given the context.
```

Comment:
minor detail: "JWT token" is redundant 😛

Reply:
Nice catch, @gustavocovas!

You're absolutely right, I'll edit that.